RenderSystems/Direct3D11/src/OgreD3D11Texture.cpp

```diff
@@ -629,7 +629,10 @@ namespace Ogre
     //---------------------------------------------------------------------
     D3D11RenderTexture::~D3D11RenderTexture()
     {
+        if (mDepthBuffer)
```

Comment:
```suggestion
        if (mDepthBuffer && PixelUtil::isDepth(mBuffer->getFormat()))
```

that should differentiate a managed mDepthBuffer from the one the texture owns, right?

Reply:
the memory contract in this code is very confusing to me -- but this does fix the leak so i'm happy if you're happy.

Reply:
the memory contract is that DepthBuffers are created and owned by the RenderSystem. This way they can be shared between RenderTargets meaning there are typically less DepthBuffers than RenderTargets.

The confusion comes in by the way I shoved in PF_DEPTH support. In this case the RenderSystem code must know that the RenderTarget handles depth itself.
The easiest way in D3D11 was give the RenderTexture its own DepthBuffer. This makes it the memory owner.

A similar argument applies to RenderWindow - however I did not write that code and the calls to RenderSystem are likely unnecessary, as the DepthBuffer is not shared either.

Reply:
ah ok that makes sense.  i verified the change fixes the leak.  thanks!